### PR TITLE
First commits

### PR DESCRIPTION
**Pull request by jsf9k**
## 🗣 Description ##

This pull request consists of the first commits for this repository.

## 💭 Motivation and context ##

We want to have a Linux instance share the EFS share via Samba so that Windows instances can mount it.  A step toward that is creating a Linux AMI with Samba installed.

## 🧪 Testing ##

All GitHub Actions pass.  I have built an AMI for our staging COOL AMI with these changes and verified that it functions as expected.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - _eschew scope creep!_
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated to reflect the changes in this PR.
- [x] All new and existing tests pass.
- [x] Build and test a staging AMI from this repository.

## ✅ Pre-merge checklist ##

- [x] Revert to the main branch of [cisagov/ansible-role-samba](https://github.com/cisagov/ansible-role-samba) once cisagov/ansible-role-samba#2 is merged.
- [x] Revert to the main branch of [cisagov/ansible-role-amazon-efs-utils](https://github.com/cisagov/ansible-role-amazon-efs-utils) once cisagov/ansible-role-amazon-efs-utils#24 is merged.
- [x] Finalize version.

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -3,7 +3,7 @@
 # These owners will be the default owners for everything in the
 # repo. Unless a later match takes precedence, these owners will be
 # requested for review when someone opens a pull request.
-* @dav3r @felddy @jsf9k @mcdonnnj
+* @dav3r @jsf9k @mcdonnnj
```
Comment (felddy) [inline .github/CODEOWNERS:6]: :(

File: src/samba.yml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+            create: no
+            option: "{{ item.option }}"
+            path: /etc/samba/smb.conf
+            section: Share
+            value: "{{ item.value }}"
+          loop:
+            - {option: "browseable", value: "yes"}
+            - {option: "guest ok", value: "yes"}
+            - {option: "guest only", value: "yes"}
+            - {option: "map to guest", value: "Bad User"}
```
Comment (felddy) [inline src/samba.yml]: I'm not sure this warrants a comment, but this seemed like a magic string to me.  Found my answer in the [`smb.conf` man page](https://www.samba.org/samba/docs/current/man-html/smb.conf.5.html): 

```
This parameter can take four different values, which tell smbd(8) what to do with user login requests that 
don't match a valid UNIX user in some way... 

Bad User - Means user logins with an invalid password are rejected, unless the username does not exist,
in which case it is treated as a guest login and mapped into the guest account.
```

**Review by felddy — APPROVED** [review]
These server message blocks are just fine.  💯

File: src/samba.yml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  name: Install Samba server
+  become: yes
+  become_method: sudo
+  roles:
+    - role: samba
+      vars:
+        create_guest_user: yes
+        server: yes
+  tasks:
+    # This way samba should not start until after the EFS share is
```
Comment (dav3r) [inline src/samba.yml]:
```suggestion
    # This way Samba should not start until after the EFS share is
```

File: src/samba.yml (hunk 2 above)
Comment (dav3r) [reply src/samba.yml]: In my opinion, that's interesting enough for a comment.

**Review by dav3r — APPROVED** [review]
Strong work! 💪 💼

File: src/samba.yml (hunk 2 above)
Comment (jsf9k) [reply src/samba.yml]: Please see commit db35b07.

File: .github/CODEOWNERS (hunk 1 above)
Comment (jsf9k) [reply .github/CODEOWNERS:6]: :violin:

**Review by mcdonnnj — APPROVED** [review]
Makes sense. Is guest access using ACL/SG configuration to ensure access control the long-term strategy?

**Comment by jsf9k** [comment]
> Makes sense. Is guest access using ACL/SG configuration to ensure access control the long-term strategy?

I was thinking so, although I hadn't fully decided on it.  The EFS mounts are _just there_ on the instance types that have them, and I don't see how guest access on the SMB share from allowed hosts is any less secure than that.  I was also hoping to avoid the headache of distributing per-assessment SMB share passwords, although I can easily generate them via Ansible [just like I do for the Bloodhound Neo4j password](https://github.com/cisagov/ansible-role-bloodhound/blob/develop/defaults/main.yml#L4-L5).

What are you thinking?

**Comment by mcdonnnj** [comment]
> > Makes sense. Is guest access using ACL/SG configuration to ensure access control the long-term strategy?
> 
> I was thinking so, although I hadn't fully decided on it. The EFS mounts are _just there_ on the instance types that have them, and I don't see how guest access on the SMB share from allowed hosts is any less secure than that. I was also hoping to avoid the headache of distributing per-assessment SMB share passwords, although I can easily generate them via Ansible [just like I do for the Bloodhound Neo4j password](https://github.com/cisagov/ansible-role-bloodhound/blob/develop/defaults/main.yml#L4-L5).
> 
> What are you thinking?

That works for me. I wanted to double-check that it was the path we were taking and make sure we had a good foundation for why it was ok to go that way in case it came up in a security review.

Also if you intended to switch to per-assessment passwords that would be something I would have to take into consideration for some of my work x)

**Pull request merged**